ipatests/pytest_ipa/integration/firewall.py

```diff
@@ -0,0 +1,127 @@
+# Authors:
+#   Thomas Woerner <twoener@redhat.com>
+#
+# Copyright (C) 2018  Red Hat
+# see file 'COPYING' for use and warranty information
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Firewall class for integration testing using firewalld"""
+
+from __future__ import absolute_import
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class Firewall:
+    def __init__(self, host):
+        """Initialize with host where firewall changes should be applied"""
+        self.host = host
+
+    def _rp_action(self, args):
```

Comment:
You could simplify some of the calls by using ``*args`` and making ``"firewall-cmd"`` a default first argument:

```
    def _rp_action(self, *args):
        cmd = ["firewall-cmd"]
        cmd.extend(args)
        self.host.run_command(cmd)
```

Reply:
Yes, will do this.